### DISCUSSION TITLE
npm install error - different versions of package webpack required?

### DISCUSSION THREAD

**Discussion by traveller195** [tomalrussell/colouring-london#841] [Q&A] [answered]
hi everybody,
unfortunately, our Colouring Germany team has some troubles with setting up the Colouring Cities environment. 
As we discussed it in the last developer call, our administrator and me could not solve the problem having different requirements (for version numbers) in the node package manager for the same package.

Maybe it is much easier, than wie excepted. Our we do something wrong in the initial steps before the install of npm...

We followed this workflow:
https://github.com/colouring-cities/colouring-london/blob/master/docs/setup-production-environment.md 
but also in the dev environment the error occured. Using Node-12.x or Node-16.x

running the npm install is not succesful: because both major versions **webpack 4** and **webpack 5** are required.

- the Colouring developers explicitly request Webpack-5.24.0
- other modules e. g..:    
    node_modules/mini-css-extract-plugin" "version": "0.9.0"
    node_modules/razzle/node_modules/file-loader" "version": "4.3.0"
    node_modules/razzle/node_modules/url-loader" "version": "2.3.0"
    node_modules/razzle/node_modules/webpack-manifest-plugin""version": "3.2.0"

need a maximum version  4 of Webpack: "webpack": "^4.x.x"  

you will find the logfiles attached.
Also with --force critical errors are occured.

also with npm ci (clean install) we could not solve this issue.
I hope you have some hints for us. 

Thanks in advance

best,

Theodor (and Joerg) from Dresden

[2022-06-13T11_32_42_530Z-debug-0.log](https://github.com/colouring-cities/colouring-london/files/8965252/2022-06-13T11_32_42_530Z-debug-0.log)
[eresolve-report.txt](https://github.com/colouring-cities/colouring-london/files/8965261/eresolve-report.txt)
[colouring_install_audit_force.txt](https://github.com/colouring-cities/colouring-london/files/8965265/colouring_install_audit_force.txt)

**Comment by traveller195** [comment]
here also logfiles with Node Version 12.14.1:

[colouring_2022-05-16T12_27_48_195Z-debug-0.log](https://github.com/colouring-cities/colouring-london/files/8967101/colouring_2022-05-16T12_27_48_195Z-debug-0.log)
[colouring_eresolve-report.txt](https://github.com/colouring-cities/colouring-london/files/8967102/colouring_eresolve-report.txt)

**Reply by matkoniecz** [reply]
Sorry for likely pointless check - but which OS / Colouring Cities / npm versions were used? (nor really expecting that one of this things will resolve issue, but maybe here something mismatches).

**Reply by traveller195** [reply]
hey :-)
just here your requested information:
OS: Ubuntu 20.04 LTS
we tried both:
prod env with version 12.x
dev env with version 16.x

the git clone was executed within the last 14 days. So, it should be a new version

thanks and greetings :-)

**Reply by matkoniecz** [reply]
Looking at it again: I see that https://github.com/colouring-cities/colouring-london/blob/master/docs/setup-dev-environment.md were recently reviewed and fully tested while https://github.com/colouring-cities/colouring-london/blob/master/docs/setup-production-environment.md was not

@tomalrussell @edwardchalstrey1 I guess that `setup-production-environment.md` should be reviewed for what should be updated/fixed and what is intentionally diverging. Large part seems to match outdated dev environment install instructions, maybe this part should be removed and only differences should be mentioned?

**Reply by edwardchalstrey1** [reply]
Yes, it's on my Todo list on my project board to test this, but at the moment other things are taking priority

**Reply by matkoniecz** [reply]
I have tested it right now on blank VM, and discovered that following [setup-dev-environment.md](https://github.com/colouring-cities/colouring-london/blob/master/docs/setup-dev-environment.md) also is running into this problem.

Bumping `mini-css-extract-plugin` to 0.12.0 appears to solve this problem and according to https://github.com/webpack-contrib/mini-css-extract-plugin/blob/master/CHANGELOG.md#0120-2020-10-07 should be save. I will test is it actually working.

**Comment by traveller195** [comment]
I start here a next reply, to copy some content from e-mail communication into this discussion:
the reason is, that Colouring Athens has implemented the Colouring Cities platform very quickly. 
So, I guess we others can learn how to avoid the described problems if we have a look to the succesful way.

George from Colouring Athens wrote:
"Hello all,

I managed to set it up using node version 12.14.1 and 10.19.0.
Node 16.x didn’t work for me as well. From the logs, I see that you are using v. 16.13.2. Can you send the error log for the version 12.14.1 as well?

Best,
George"

I just want to ask about some basic facts of your succesful set up (maybe we can adapt from your good experience):

1. which OS did you use? if Ubuntu, which version?
2. did you use a dockerized environment, our just the "old school" way of installing everything without containers?
3. did you implement the dev or the production environment?
4. could you maybe send me the URL of the described workflow, you applied? (URL for the docs within the colouring london/ cities repo?

so maybe then, it will be easier to understand, where the problems are.

Theodor, Dresden (Germany)

**Reply by gpanag** [reply]
1. Ubuntu 20.04 LTS
2. old school 😉 
3. Dev with Node 10.19.0, prod with Node 12.14.1.
4. [setup-production-environment.md](https://github.com/colouring-cities/colouring-london/files/8994898/setup-production-environment.md)
[setup-dev-environment.md](https://github.com/colouring-cities/colouring-london/files/8994899/setup-dev-environment.md)
but on the production environment we used nvm to install node.

I remember having some issues, but all version related. Can you also check the npm version? I suppose it shouldn't be a problem, but just to be sure. We use npm 6.13.4 with Node 12.14.1.

**Comment by matkoniecz** [comment, marked as answer]
It is supposed to be fixed by https://github.com/colouring-cities/colouring-london/pull/851 - can you test/review?

And thanks for reporting the problem!

**Reply by edwardchalstrey1** [reply]
Thanks for looking at this @matkoniecz :)

**Reply by traveller195** [reply]
thanks @matkoniecz and all of you. Now, your fixed docs works for the npm install. As @tomalrussell mentioned in https://github.com/colouring-cities/colouring-london/pull/851 there are still warnings, but no further errors while installing npm

thanks :)
